tools/integration_tests/log_content/helper.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for write large files sequentially and randomly.
+package log_content
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+)
+
+const (
+	MiB      = 1024 * 1024
+	FileName = "fileName.txt"
+)
+
+var (
+	logFileOffset int
+)
+
+func extractRelevantLogsFromLogFile(t *testing.T, logFile string, logFileOffset *int) (logString string) {
+	// Read the entire log file at once. This can be optimized by reading
+	// a bunch of lines at once, then eliminating the found
+	// expected substrings one by one.
+	bytes, err := os.ReadFile(logFile)
+	if err != nil {
+		t.Errorf("Failed in reading logfile %q: %v", logFile, err)
+	}
+	completeLogString := string(bytes)
+
+	logString = completeLogString[*logFileOffset:]
+	*logFileOffset = len(completeLogString)
```

Comment:
I didnt follow this line. what is logFileOffset used for?

Reply:
logFileOffset is the offset from which the logs for the current test start. This is needed as log-file is shared across all tests in the package and we should search the expected/unexpected substrings only in the logs generated by the current test.

Reply:
@vadlakondaswetha ^

Reply:
This is not correct. If the same log file is used for every test then at the begining of the test you need the set of the offset of the log file as apart of test initialization. You shouldn't update the offset in a function and expect that function to be called in the previous test.